src/dune/build_system.ml

```diff
@@ -1531,6 +1531,9 @@ end = struct
                     f
                 ];
               false
+            | exception Sys_error m ->
+              Log.info [ Pp.textf "error retrieving data file: %s" m ];
```

Comment:
Not entirely related to Sys_error, but reading this code again I'm left asking: is it ok to stop retrieving the other files in the cache just because we encountered one failure?

Reply:
Since it's per-rule, if one single target file is missing / cannot be retrieved from the cache, we will have to execute the rule anyway, so the idea is we might as well stop asap and skip to running the rule.

I'll merge this PR since it is strictly an improvement, let's open an issue for this discussion if needed.